app/src/main/java/com/mxt/anitrend/util/collection/GroupingUtil.kt

```diff
@@ -171,6 +172,39 @@ object GroupingUtil {
         return getDifference(model, entityMap)
     }
 
+    /**
+     * Groups characters by year
+     * <br></br>
+     *
+     *
+     * @param edges The potential external model response which needs to be grouped
+     * @param model The current model item/s containing all data minus current mediaItems
+     */
+    fun groupCharactersByYear(edges: List<MediaEdge>, model: List<RecyclerItem>?): List<RecyclerItem> {
+        val entityMap = if (!CompatUtil.isEmpty(model)) ArrayList(model!!) else ArrayList()
```

Comment:
Kotlin offers some extension function for checking nullability for lists and or converting into a mutable, you could use:
```kotlin
// The resulting type would be MutableList<RecyclerItem>
val entityMap = model.orEmpty().toMutableList()
```

Reply:
Maybe that should be a separate change as this exact line is used 6 times here.

Reply:
You're right, Initially the CompatUtils especially the `List` methods were meant used for java this was before kotlin was even a thing and I never got to refactor everything to kotlin only, so In that case we can just ignore this 💯